MacOS wheels: curses libs not statically linked

Currently MacOS wheels are broken. When running `nrnivmodl` we get: 
```
clang: error: no such file or directory: '/opt/nrnwheel/ncurses/lib/libcurses.a'
clang: error: no such file or directory: '/opt/nrnwheel/ncurses/lib/libform.a'
INFO : Using neuron-nightly Package (Developer Version)
make: *** [mech_lib_shared] Error 1
Traceback (most recent call last):
  File "/Users/runner/work/nrn-build-ci/nrn-build-ci/nrn/nrn_test_venv_39/bin/nrnivmodl", line 66, in <module>
    subprocess.check_call([exe, *sys.argv[1:]])
  File "/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/Users/runner/work/nrn-build-ci/nrn-build-ci/nrn/nrn_test_venv_39/lib/python3.9/site-packages/neuron/.data/bin/nrnivmodl', 'tmp_mod']' returned non-zero exit status 2.
```
Even though #1371 got in several weeks ago, we've only started to see this now because wheel publishing wasn't possible (#1389)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,8 +189,12 @@ if(READLINE_FOUND)
     # define for src/oc/hoc.cpp
     set(DEF_RL_GETC_FUNCTION use_rl_getc_function)
   endif()
-  find_package(Curses QUIET)
-  find_package(Termcap QUIET)
+  # If we are not using self contained, static readline library
+  # created for building wheel then only look for curses and termcap
+  if(NOT NRN_WHEEL_STATIC_READLINE)
+    find_package(Curses QUIET)
+    find_package(Termcap QUIET)
+  endif()
 endif()
 
 # Reset original CMake library suffixes

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -97,18 +97,12 @@ jobs:
       cmake --version
     displayName: 'Install OSX System Dependencies'
 
-  # readline & ncurses have been manually built with MACOSX_DEPLOYMENT_TARGET=10.9 and stored as secure files on Azure.
+  # readline has been manually built with MACOSX_DEPLOYMENT_TARGET=10.9 and stored as secure files on Azure.
   # See `packaging/python/Dockerfile` for build instructions.
   #
   # Secure files documentation:
   #   https://docs.microsoft.com/en-us/azure/devops/pipelines/library/secure-files?view=azure-devops
   #   NOTE: when uploading new secure files, access must be permitted from pipeline interface (see message there)
-  - task: DownloadSecureFile@1
-    name: ncursesSF
-    displayName: 'Download ncurses secure file'
-    inputs:
-      secureFile: 'ncurses6.2.tar.gz'
-
   - task: DownloadSecureFile@1
     name: readlineSF
     displayName: 'Download readline secure file'
@@ -121,7 +115,6 @@ jobs:
       export SDKROOT=$(xcrun --sdk macosx --show-sdk-path)
       export NRN_BUILD_FOR_UPLOAD=1
       sudo mkdir /opt/nrnwheel
-      sudo tar -zxf $(ncursesSF.secureFilePath) --directory /opt/nrnwheel/
       sudo tar -zxf $(readlineSF.secureFilePath) --directory /opt/nrnwheel/
       packaging/python/build_wheels.bash osx $(python.version)
     displayName: 'Build MacOS Wheel'

